mmdet/datasets/pipelines/ipu_pipelines.py

```diff
@@ -0,0 +1,239 @@
+
+
+@PIPELINES.register_module()
+class IPUCollect:
+    """IPU version Collect.
+
+    The additional feature is:
+        1. meta_tensor_keys specify tensors which will be converted to
+        torch.Tensor
+        2. remove_img_meta determines whether remove img_metas from results
+    Args:
+        keys (Sequence[str]): Keys of results to be collected in ``data``.
+        meta_keys (Sequence[str], optional): Meta keys to be converted to
+            ``mmcv.DataContainer`` and collected in ``data[img_metas]``.
+            Default: ``('filename', 'ori_filename', 'ori_shape', 'img_shape',
+            'pad_shape', 'scale_factor', 'flip', 'flip_direction',
+            'img_norm_cfg')``
+        meta_tensor_keys (tuple[str]): keys of data which will be converted
+            to torch.Tensor
+        remove_img_meta (bool): whether remove img_metas from results or not.
+    """
+
+    def __init__(self,
+                 keys,
+                 meta_keys=('filename', 'ori_filename', 'ori_shape',
+                            'img_shape', 'pad_shape', 'scale_factor', 'flip',
+                            'flip_direction', 'img_norm_cfg'),
+                 meta_tensor_keys=tuple(),
+                 remove_img_meta=False):
+        self.keys = keys
+        self.meta_keys = meta_keys
+        self.meta_tensor_keys = meta_tensor_keys
+        self.remove_img_meta = remove_img_meta
+        self.shape_dic = {}
+
+    def check_shape(self, name, shape):
+        if name in self.shape_dic:
+            org_shape = self.shape_dic[name]
+            assert shape == org_shape, 'Tensor shape changed!'
+        else:
+            self.shape_dic[name] = shape
+
+    def __call__(self, results):
+        data = {}
+        img_meta = {}
+        for key in self.meta_keys:
+            if key in self.meta_tensor_keys:
+                img_meta[key] = torch.from_numpy(results[key])
+            else:
+                img_meta[key] = results[key]
+        if self.remove_img_meta:
+            data['img_metas'] = DC(img_meta, cpu_only=True)
+        for key in self.keys:
+            data[key] = results[key]
+            self.check_shape(key, results[key].data.shape)
+        return data
+
+    def __repr__(self):
+        return self.__class__.__name__ + \
+               f'(keys={self.keys}, meta_keys={self.meta_keys})'
+
+
+@PIPELINES.register_module()
+class GetTargetsOutsideForYolo:
+    """Make anchors and match them with groud truth in YoloV3's method.
+
+    Args:
+        featmap_sizes (list[int,]): Static featmap size of output tensor
+            of yolo's head.
+        num_levels (int): number of feature levels.
+    """
+
+    def __init__(self, featmap_sizes, num_levels=3):
+        s1, s2, s3 = featmap_sizes
+        self.featmap_sizes = [
+            torch.Size([s1, s1]),
+            torch.Size([s2, s2]),
+            torch.Size([s3, s3])
+        ]
+        self.num_levels = num_levels
+        self.model = None
```

Comment:
We may add comments to indicate where we will make it a real model so that we can use it to do target assignment later.